core/blockchain_reader.go

```diff
@@ -230,9 +230,9 @@ func (bc *BlockChain) GetBlobSidecarsByNumber(number uint64) *types.BlobSidecars
 	if sidecars, ok := bc.blobSidecarsCache.Get(hash); ok {
 		return sidecars.(*types.BlobSidecars)
 	}
-	
+
 	sidecars := rawdb.ReadBlobSidecars(bc.db, hash, number)
-	bc.blobSidecarsCache.Add(hash, sidecars)
+	bc.blobSidecarsCache.Add(hash, &sidecars)
```

Comment:
Why do we need this change?

Reply:
The `sidecars` is retrieved as follow, so it should be added as pointer, otherwise it can't be cast.
```
if sidecars, ok := bc.blobSidecarsCache.Get(hash); ok {
	return sidecars.(*types.BlobSidecars)
}
```